### DISCUSSION TITLE
NestJS, NextJS and Custom Login UI

### DISCUSSION THREAD

**Discussion by ilterugur** [zitadel/zitadel#7343] [Q&A]
Hello,
I've read many Zitadel official docs pages and GitHub repo instructions but am unsure about the best practices in my case.

- I will use NestJS on backend and NextJS (app directory) on frontend
- I will use MongoDB on NestJS and I need to "bind" user information on my DB with Zitadel's db
- Registration, login and user info updating processes need to be done on my NextJS frontend with a custom UI. I need to handle all these scenarios on the same NextJS app
- I need to develop a NestJS authentication guard which connects to Zitadel so I need to detect which Zitadel user is sending requests through NextJS frontend to my NestJS backend
- In the future, I will need to develop a React Native app for this project

How can I use my NestJS backend as a bridge between NextJS frontend and Zitadel? Which auth types should I configure on Zitadel Panel?

I need some suggestions about this case. Are there any example monorepos or flows for this case? Thanks

**Comment by hifabienne** [comment]
hei @ilterugur 
We are currently also building a new login UI using our new session APIs.
Its also build with Next.js
It is not finished yet, but I think you will already get some informations there:
https://github.com/zitadel/typescript

@peintnermax Is probably able to answer your questions better.

**Reply by ilterugur** [reply]
Thanks for your response,
I've already seen typescript repo but since you've mentioned it too, I guess I should focus on that repo as a reference for my project.
It looks like I need to use zitadel-server package. Have you tried to use this package on a NestJS server? A basic example would be really useful if there is any.
